float32[] position6D
--------------------------------
float32[] angles